terraform/main.tf

```
@@ -301,6 +332,213 @@ resource "aws_iam_role_policy" "api_task-publish_to_arpa_audit_report_queue" {
+    principal_identifier = "ecs-tasks.amazonaws.com"
+    source_arn           = module.api.ecs_service_arn
+  }
+  sqs_max_receive_count             = 2
+  sqs_dlq_message_retention_seconds = 1209600 # 14 days, in seconds
+
+  # Logging
+  log_retention = var.api_log_retention_in_days
+
+  # Secrets
+  ssm_path_prefix = var.ssm_service_parameters_path_prefix
+
+  # Postgres
+  rds_db_connect_resources = module.postgres.rds_db_connect_resources_list
+  postgres_username        = module.postgres.master_username
+  postgres_endpoint        = module.postgres.cluster_endpoint
+  postgres_port            = module.postgres.cluster_port
+  postgres_db_name         = module.postgres.default_db_name
+}
+
+data "aws_iam_policy_document" "publish_to_digest_email_send_queue" {
+  statement {
+    sid       = "AllowPublishToQueue"
+    actions   = ["sqs:SendMessage"]
+    resources = [module.digest_email_send.sqs_queue_arn]
+  }
+}
+
+resource "aws_iam_role_policy" "digest_email_get_grants-publish_to_digest_email_send_queue" {
+  name_prefix = "send-digest-email-send-requests"
+  role        = module.digest_email_get_grants.ecs_task_role_name
+  policy      = data.aws_iam_policy_document.publish_to_digest_email_send_queue.json
+}
+
+module "digest_email_send" {
+  source                   = "./modules/sqs_consumer_task"
+  namespace                = "${var.namespace}-digest_email_send"
+  permissions_boundary_arn = local.permissions_boundary_arn
+
+  # Networking
+  subnet_ids         = local.private_subnet_ids
+  security_group_ids = [module.digest_email_send_security_group.id]
+
+  # Task configuration
+  ecs_cluster_name      = join("", aws_ecs_cluster.default.*.name)
+  docker_tag            = var.api_container_image_tag
+  unified_service_tags  = local.unified_service_tags
+  stop_timeout_seconds  = 120
+  consumer_task_command = ["node", "./src/lib/digest/sendEmail.js"]
+  consumer_container_environment = {
+    API_DOMAIN          = local.api_domain_name
+    DATA_DIR            = "/var/data"
+    NODE_OPTIONS        = "--max_old_space_size=3584" # Reserve 512 MB for other task resources
+    NOTIFICATIONS_EMAIL = "grants-notifications@${var.website_domain_name}"
+    WEBSITE_DOMAIN      = "https://${var.website_domain_name}"
+  }
+  consumer_task_efs_volume_mounts = [{
+    name            = "data"
+    container_path  = "/var/data"
+    read_only       = false
+    file_system_id  = module.api.efs_data_volume_id
+    access_point_id = module.api.efs_data_volume_access_point_id
+  }]
+  additional_task_role_json_policies = {
+    send-emails = module.api.send_emails_policy_json
+  }
+
+  # Task resource configuration
+  # TODO: Tune these values after observing usage in different environments.
+  #       See also: --max_old_space_size in NODE_OPTIONS env var.
+  consumer_task_size = {
+    cpu    = 1024 # 1 vCPU
+    memory = 4096 # 4 GB
+  }
+
+  # Messaging
+  autoscaling_message_thresholds = [1, 3, 5, 10, 20, 50]
```

This ultimately depends on anticipated number of individual emails that will get queued up; try this as a starting point:

```suggestion
  autoscaling_message_thresholds = [100, 200, 400, 600, 800, 1000, 5000, 10000]
```